src/table/iterator.rs

```diff
@@ -187,6 +200,261 @@ impl BlockIterator {
+                Greater => true,
+                _ => false,
+            }
+        });
+
+        if idx == 0 {
+            self.seek_helper(0, key);
+            return;
+        }
+
+        self.seek_helper(idx - 1, key);
+        if IteratorError::check_eof(&self.err) {
+            if idx == self.table.as_ref().offsets_length() {
+                return;
+            }
+            self.seek_helper(idx, key);
+        }
+    }
+
+    // seek_inner will reset iterator and seek to >= key.
+    fn seek_inner(&mut self, key: &Bytes) {
+        self.seek_from(key, SeekPos::Origin);
+    }
+
+    // seek_for_prev will reset iterator and seek to <= key.
+    fn seek_for_prev(&mut self, key: &Bytes) {
+        self.seek_from(key, SeekPos::Origin);
+        if self.key() != key {
+            self.prev_inner();
+        }
+    }
+
+    fn next_inner(&mut self) {
+        self.err = None;
+
+        if self.bpos > self.table.as_ref().offsets_length() {
+            self.err = Some(IteratorError::EOF);
+            return;
+        }
+
+        if BlockIterator::require_load(&self.block_iterator) {
+            match self.table.as_ref().block(self.bpos, self.use_cache()) {
+                Ok(block) => {
+                    let block_iterator = self.get_block_iterator(block);
+                    block_iterator.seek_to_first();
+                    self.err = block_iterator.err.clone();
+                }
+                Err(err) => self.err = Some(err.into()),
+            }
+        } else {
+            let bi = self.block_iterator.as_mut().unwrap();
+            bi.next();
+            if !bi.valid() {
+                self.bpos += 1;
+                bi.data.clear();
+                self.next_inner();
+            }
+        }
+    }
+
+    fn prev_inner(&mut self) {
+        self.err = None;
+
+        if BlockIterator::require_load(&self.block_iterator) {
+            match self.table.as_ref().block(self.bpos, self.use_cache()) {
+                Ok(block) => {
+                    let block_iterator = self.get_block_iterator(block);
+                    block_iterator.seek_to_last();
+                    self.err = block_iterator.err.clone();
+                }
+                Err(err) => self.err = Some(err.into()),
+            }
+        } else {
+            let bi = self.block_iterator.as_mut().unwrap();
+            bi.prev();
+            if !bi.valid() {
+                if self.bpos == 0 {
+                    self.bpos = std::usize::MAX;
+                    // At this point, the iterator must be reset before using.
+                    // If the user calls `next`, `bpos` will overflow and panic.
+                    self.err = Some(IteratorError::EOF);
+                    return;
+                }
+                self.bpos -= 1;
+                bi.data.clear();
+                self.prev_inner();
+            }
+        }
+    }
+
+    pub fn key(&self) -> &[u8] {
```

Comment:
How about returning `&Bytes`?

Reply:
As `key` is stores as `BytesMut`, returning `Bytes` requires a clone.

Reply:
So I think it's better to decide whether to clone when using this key, instead of always cloning it when getting the key.